src/main/scala/firrtl/Mappers.scala

```diff
@@ -79,7 +79,10 @@ object Mappers {
           case s: DefWire => DefWire(s.info,f(s.name),s.tpe)
           case s: DefPoison => DefPoison(s.info,f(s.name),s.tpe)
           case s: DefRegister => DefRegister(s.info,f(s.name), s.tpe, s.clock, s.reset, s.init)
-          case s: DefMemory => DefMemory(s.info,f(s.name), s.data_type, s.depth, s.write_latency, s.read_latency, s.readers, s.writers, s.readwriters)
+          case s: DefMemory =>
+            DefMemory(s.info,f(s.name), s.data_type, s.depth, s.write_latency,
```

Comment:
Yeah that's a good point. 

I originally did that to make mapping _ to __ easier. Regardless, I think you're right that String -> String mapping does not imply changing the readers and writers. Also, I no longer believe _ -> __ is the right approach to dealing with name collisions anyway.